services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouteWithSubnetMapperServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.route.service.Impl;
+
+import com.futurewei.alcor.common.db.CacheException;
+import com.futurewei.alcor.common.exception.DatabasePersistenceException;
+import com.futurewei.alcor.common.exception.ResourceNotFoundException;
+import com.futurewei.alcor.common.exception.ResourcePersistenceException;
+import com.futurewei.alcor.route.dao.RouteRepository;
+import com.futurewei.alcor.route.dao.RouteWithSubnetMapperRepository;
+import com.futurewei.alcor.route.service.RouteWithSubnetMapperService;
+import com.futurewei.alcor.web.entity.route.RouteEntity;
+import com.futurewei.alcor.web.entity.route.SubnetToRouteMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class RouteWithSubnetMapperServiceImpl implements RouteWithSubnetMapperService {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    RouteWithSubnetMapperRepository routeWithSubnetMapperRepository;
+
+    @Autowired
+    RouteRepository routeRepository;
+
+    @Override
+    public SubnetToRouteMapper getBySubnetId(String subnetId) throws ResourceNotFoundException, ResourcePersistenceException {
```

Comment:
quick question: why do we need two get methods,
getBySubnetId => to get SubnetToRouteMapper
getRuleBySubnetId => to get the real RouteEntity List

Why do we need the first get method?

Reply:
I think I dont have getBySubnetId, just getRulesBySubnetId

Reply:
Oh I see ~ that's two different method, the one is to get SubnetToRouteMapper entity, the other one is to get List<RouteEntity>

Reply:
The one is to get SubnetToRouteMapper entity, the other one is to get List<RouteEntity>